plugin/ale.vim

```diff
@@ -87,6 +87,9 @@ let g:ale_lint_on_save = get(g:, 'ale_lint_on_save', 1)
 " This flag can be set to 1 to enable linting when the filetype is changed.
 let g:ale_lint_on_filetype_changed = get(g:, 'ale_lint_on_filetype_changed', 1)
 
+" This flag can be set to 1 to enable linting when vim gains focus.
+let g:ale_lint_on_focus_gained = get(g:, 'ale_lint_on_focus_gained', 1)
```

Comment:
Document the option and turn it off by default. There are thousands of people using ALE, and many of them will have told ALE not to lint their buffers when they enter a file. They will be surprised if it starts linting their buffers when Vim regains focus after they update ALE.

Reply:
Hm... maybe we should use an existent check `g:ale_lint_on_enter`, what do you think? Not sure adding an extra option is worth it considering the semantics of "entering" is so poorly defined.

Reply:
Also — maybe we can make such behaviour configurable per-linter?

Reply:
I think a new option is appropriate, but it should be off by default.